### PR TITLE
MGMT-2714 Collect must-gather

### PR DESCRIPTION
**Pull request by YuviGold**

### CODE REVIEW

File: Dockerfile.test-infra [hunk 1]
```diff
@@ -14,12 +14,12 @@ RUN mkdir -p /root/.terraform.d/plugins
 RUN curl -SL https://github.com/dmacvicar/terraform-provider-libvirt/releases/download/v0.6.2/terraform-provider-libvirt-0.6.2+git.1585292411.8cbe9ad0.Fedora_28.x86_64.tar.gz | \
   tar -xz -C /root/.terraform.d/plugins
 
-# RUN curl -SL https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/4.6.0-rc.4/openshift-client-linux-4.6.0-rc.4.tar.gz | tar -xz -C /usr/bin
+RUN curl -SL https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/4.6.0/openshift-client-linux-4.6.0.tar.gz | tar -xz -C /usr/bin
```
Comment (tsorya) [inline Dockerfile.test-infra:17]: didn't it fail?

File: discovery-infra/download_logs.py [hunk 2]
```diff
@@ -102,6 +112,12 @@ def get_ui_url_from_api_url(api_url: str):
         raise KeyError(api_url)
 
 
+def download_must_gather(kubeconfig: str, dest_dir: str):
+    log.info(f"Downloading must-gather to {dest_dir}")
+    command = f"oc --insecure-skip-tls-verify --kubeconfig={kubeconfig} adm must-gather --dest-dir {dest_dir} > {dest_dir}/must-gather.log"
```
Comment (tsorya) [inline discovery-infra/download_logs.py:117]: one more time oc command? Please talk to @RazRegev and use oc utils

File: discovery-infra/download_logs.py [hunk 3]
```diff
@@ -63,6 +66,13 @@ def download_logs(client: InventoryClient, cluster: dict, dest: str):
     with suppress(assisted_service_client.rest.ApiException):
         client.download_cluster_logs(cluster['id'], os.path.join(output_folder, f"cluster_{cluster['id']}_logs.tar"))
 
+    kubeconfig_path = os.path.join(output_folder, "kubeconfig")
+
+    with suppress(assisted_service_client.rest.ApiException):
+        if cluster['status'] in [ClusterStatus.INSTALLED]:
+            client.download_kubeconfig(cluster['id'], kubeconfig_path)
+            download_must_gather(kubeconfig_path, os.path.join(output_folder, "must-gather"))
```
Comment (tsorya) [inline discovery-infra/download_logs.py:74]: and in finalizing? or if we failed cause of finalizing timeout?

Comment (YuviGold) [reply discovery-infra/download_logs.py:74]: Can we gather must-gather on finalizing?

File: Dockerfile.test-infra (hunk 1 above)
Comment (YuviGold) [reply Dockerfile.test-infra:17]: They published a stable 4.6.0 version, not rc

File: discovery-infra/download_logs.py (hunk 2 above)
Comment (YuviGold) [reply discovery-infra/download_logs.py:117]: Yeah

File: discovery-infra/download_logs.py (hunk 3 above)
Comment (tsorya) [reply discovery-infra/download_logs.py:74]: yes

File: scripts/assisted_deployment.sh [hunk 4]
```diff
@@ -19,6 +19,11 @@ function set_dns() {
     echo "server=/api.${CLUSTER_NAME}-${NAMESPACE}.${BASE_DOMAIN}/${API_VIP}" | sudo tee -a /etc/NetworkManager/dnsmasq.d/openshift-${CLUSTER_NAME}.conf
     sudo systemctl reload NetworkManager
 
+    # Change default API VIP with DHCP chosen one
+    API_VIP=$(sudo virsh net-dhcp-leases test-infra-net-${NAMESPACE} | grep api | awk '{print $5}' | cut -d"/" -f1)
+    virsh net-update test-infra-net-${NAMESPACE} delete dns-host "<host ip='192.168.126.100'><hostname>api.${CLUSTER_NAME}-${NAMESPACE}.${BASE_DOMAIN}</hostname></host>"
```
Comment (tsorya) [inline scripts/assisted_deployment.sh:24]: why u need this?

Comment (tsorya) [reply scripts/assisted_deployment.sh:24]: u should do it here, it can be part of creating vms

**Pull request merged**